ext/pdo_oci/tests/pdo_oci_stmt_getcolumnmeta.phpt

```diff
@@ -0,0 +1,281 @@
+			return false;
+		}
+
+		$stmt = $db->prepare('SELECT id, label FROM test');
+		$stmt->execute();
+		$meta = $stmt->getColumnMeta(1);
+		$row = $stmt->fetch(PDO::FETCH_ASSOC);
+
+		if (empty($meta)) {
+			printf("[%03d + 2] getColumnMeta() failed, %d - %s\n", $offset,
+				$stmt->errorCode(), var_export($stmt->errorInfo(), true));
+			return false;
+		}
+		$elements = array('flags', 'name', 'len', 'precision', 'pdo_type');
+		foreach ($elements as $k => $element)
+			if (!isset($meta[$element])) {
+				printf("[%03d + 3] Element %s missing, %s\n", $offset,
+					$element, var_export($meta, true));
+				return false;
+			}
+
+		if (!is_null($native_type)) {
+			if (!isset($meta['native_type'])) {
+				printf("[%03d + 4] Element native_type missing, %s\n", $offset,
+					var_export($meta, true));
+				return false;
+			}
+
+			if (!is_array($native_type))
+				$native_type = array($native_type);
+
+			$found = false;
+			foreach ($native_type as $k => $type) {
+				if ($meta['native_type'] == $type) {
+					$found = true;
+					break;
+				}
+			}
+
+			if (!$found) {
+				printf("[%03d + 5] Expecting native type %s, %s\n", $offset,
+					var_export($native_type, true), var_export($meta, true));
+				return false;
+			}
+		}
+
+		if (!is_null($pdo_type) && ($meta['pdo_type'] != $pdo_type)) {
+			printf("[%03d + 6] Expecting PDO type %s got %s (%s)\n", $offset,
+				$pdo_type, var_export($meta, true), var_export($meta['native_type']));
+			return false;
+		}
+
+		return true;
+	}
+
+	echo "Test 2.2 testing numeric columns\n";
+
+	test_meta($db, 20, 'NUMBER'         , 0                    , 'NUMBER', PDO::PARAM_STR);
+	test_meta($db, 30, 'NUMBER'         , 256                  , 'NUMBER', PDO::PARAM_STR);
+	test_meta($db, 40, 'INT'            , 256                  , 'NUMBER', PDO::PARAM_STR);
+	test_meta($db, 50, 'INTEGER'        , 256                  , 'NUMBER', PDO::PARAM_STR);
+	test_meta($db, 60, 'NUMBER'         , 256.01               , 'NUMBER', PDO::PARAM_STR);
+	test_meta($db, 70, 'NUMBER'         , -8388608             , 'NUMBER', PDO::PARAM_STR);
+
+	test_meta($db, 80, 'NUMBER'         , 2147483648           , 'NUMBER', PDO::PARAM_STR);
+	test_meta($db, 90, 'NUMBER'         , 4294967295           , 'NUMBER', PDO::PARAM_STR);
+
+	test_meta($db, 100, 'DEC'             , 1.01               , 'NUMBER'       , PDO::PARAM_STR);
+	test_meta($db, 110, 'DECIMAL'         , 1.01               , 'NUMBER'       , PDO::PARAM_STR);
+	test_meta($db, 120, 'FLOAT'           , 1.01               , 'FLOAT'        , PDO::PARAM_STR);
+	test_meta($db, 130, 'DOUBLE PRECISION', 1.01               , 'FLOAT'        , PDO::PARAM_STR);
+	test_meta($db, 140, 'BINARY_FLOAT'    , 1.01               , 'BINARY_FLOAT' , PDO::PARAM_STR);
+	test_meta($db, 150, 'BINARY_DOUBLE'   , 1.01               , 'BINARY_DOUBLE', PDO::PARAM_STR);
+
+	echo "Test 2.3 testing temporal columns\n";
+
+	test_meta($db, 160, 'DATE'           , '2008-04-23'        , 'DATE', PDO::PARAM_STR);
```

Comment:
This causes a diff in locales like the USA.   Try by setting:
```
export NLS_LANG=.utf8  # or whatever you like - this var is needed otherwise the next isn't looked up
export NLS_DATE_FORMAT=DD-MON-YY
make test TESTS=ext/pdo_oci/tests/pdo_oci_stmt_getcolumnmeta.phpt
```
This fails with:
```
Warning: PDO::exec(): SQLSTATE[HY000]: General error: 1861 OCIStmtExecute: ORA-01861: literal does not match format string
```

One solution is to wrap the insert with a TO_DATE (which may be tricky since your INSERT forces quotes around the data):
```
SQL> insert into t values ('2018-09-29');
insert into t values ('2018-09-29')
                      *
ERROR at line 1:
ORA-01861: literal does not match format string

SQL> insert into t values (TO_DATE('2018-09-29', 'YYYY-MM-DD'));

1 row created.

SQL> 
```

But it may just be easier to force the connection to the given format by executing this before the rest of the test:
```
	$db->exec("alter session set nls_date_format='YYYY-MM-DD'");
```